facebook_chat_statistics.py

```diff
@@ -17,16 +17,21 @@
 you = 'You'
 partner = 'Partner'
 
+# Convert unicode characters into what Python expects them to look like
+for message in data['messages']:
+	message['sender_name'] = message['sender_name'].encode('raw_unicode_escape').decode('utf-8')
+	if 'content' in message:
+		message['content'] = message['content'].encode('raw_unicode_escape').decode('utf-8')
+
+# Sort the messages by time
+data['messages'] = sorted(data['messages'], key=lambda message: message['timestamp'])
```

Comment:
Seems unnecessary to sort such a large array when it already is sorted by time. I would rather see changes in the fetch timeline data code so that it worked with the message order given by Facebook.

Reply:
The array is not sorted by time, but I've removed the sort in favor of doing a min/max to find the start and end dates and fixing the timeline code to properly move back and forth through the timeline/nbr_times_day arrays.

Reply:
My data which I downloaded two days ago is sorted by time (newest first). I have made a new [branch](https://github.com/davidkrantz/FacebookChatStatistics/tree/updated-format) showing my proposal, which mostly consists of your first commit. I get same result when I run my new code and when I run both of your commits, except mine is little faster. Please try it yourself and get back to me.

Reply:
I just re-downloaded an archive and it is not in order (as you said, newest first) so you can go with your branch.

Reply:
If your message data still is not in order, this needs further investigation. If the order somehow varies with the user my solution would not always work, while your's would.

Reply:
Sorry, that was a typo. The newly downloaded one is in order, the older one is not.

Reply:
Aha ok, great! If you would like you can copy my branch solution and commit the changes here so that I can merge this PR. In that way you will get cred for your contribution.

Reply:
Good idea, I just did it.